controllers/pipelines/internal/provider/predicates/service_predicate.go

```diff
@@ -0,0 +1,26 @@
+package predicates
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+type ServiceChangedPredicate struct {
+	predicate.Funcs
+}
+
+// Create Used to filter out create events triggered by a controller operation
+func (ServiceChangedPredicate) Create(e event.CreateEvent) bool {
+	svc, ok := e.Object.(*corev1.Service)
+	if !ok {
+		return true
+	}
+
+	//Ignore create events caused by controller ops
+	if svc.Annotations != nil && svc.Annotations[ControllerManagedKey] == "true" {
+		return false
+	}
+
+	return true
```

Comment:
can be simplified same as above

Reply:
removed this code block within [a2f6bc4](https://github.com/sky-uk/kfp-operator/pull/476/commits/a2f6bc4c820a0838497ec6c03768d77762610472)